@xen-orchestra/web/src/pages/vm/new.vue

```diff
@@ -0,0 +1,924 @@
+import type { XoVmTemplate } from '@/types/xo/vm-template.type'
+import type { Branded } from '@core/types/utility.type'
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import VtsResource from '@core/components/resources/VtsResource.vue'
+import VtsResources from '@core/components/resources/VtsResources.vue'
+import VtsTable from '@core/components/table/VtsTable.vue'
+import UiButton from '@core/components/ui/button/UiButton.vue'
+import UiButtonIcon from '@core/components/ui/button-icon/UiButtonIcon.vue'
+import UiCard from '@core/components/ui/card/UiCard.vue'
+import UiCheckbox from '@core/components/ui/checkbox/UiCheckbox.vue'
+import UiCheckboxGroup from '@core/components/ui/checkbox-group/UiCheckboxGroup.vue'
+import UiHeadBar from '@core/components/ui/head-bar/UiHeadBar.vue'
+import UiInput from '@core/components/ui/input/UiInput.vue'
+import UiLabel from '@core/components/ui/label/UiLabel.vue'
+import UiRadioButton from '@core/components/ui/radio-button/UiRadioButton.vue'
+import UiTextarea from '@core/components/ui/text-area/UiTextarea.vue'
+import UiTitle from '@core/components/ui/title/UiTitle.vue'
+import UiToaster from '@core/components/ui/toaster/UiToaster.vue'
+import {
+  faAlignLeft,
+  faAngleDown,
+  faAt,
+  faDatabase,
+  faDisplay,
+  faMemory,
+  faMicrochip,
+  faNetworkWired,
+  faPlus,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+
+import { computed, reactive, ref, watch } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { useRouter } from 'vue-router'
+
+// i18n setup
+const { t } = useI18n()
+const router = useRouter()
+
+// Toaster
+const errorMessage = ref('')
+const isOpen = ref(false)
+
+const isBusy = ref(false)
+
+const { records: networks, get: getNetwork } = useNetworkStore().subscribe()
+const { getPifsByNetworkId, get: getPif } = usePifStore().subscribe()
+const { records: pools } = usePoolStore().subscribe()
+const { vmsTemplatesByPool } = useVmTemplateStore().subscribe()
+const { records: srs, get: getSr, vdiIsosBySrName } = useSrStore().subscribe()
+const { get: getVbd } = useVbdStore().subscribe()
+const { get: getVdi } = useVdiStore().subscribe()
+const { get: getVif } = useVifStore().subscribe()
+const { hostsByPool } = useHostStore().subscribe()
+
+// INSTALL METHOD
+type InstallMethod = 'no-config' | 'ssh-key' | 'custom_config' | 'cdrom' | 'network' | undefined
+
+interface InstallMode {
+  method: InstallMethod
+  repository: string
+}
+
+const useInstallMode = () => {
+  const installMode = reactive<InstallMode>({
+    method: undefined,
+    repository: '',
+  })
+
+  const resetInstallMethod = () => {
+    installMode.method = undefined
+    installMode.repository = ''
+  }
+
+  const setInstallMethod = (method: InstallMethod) => {
+    resetInstallMethod()
+    installMode.method = method
+
+    if (method === 'network') {
+      installMode.repository = ' '
```

Comment:
Why did you add a space in the empty string?

Reply:
because an error is thrown: 'required min lenght: 1'. To be fixed in the REST-API